pharmaverse/aNCA
# Enhancement: setup system for TLG (plots)

## Issue

Closes #53

## Description

- Introduces new module and specification system allowing for converting simple functions that generate plots into dynamic plot display that allows the user to customize plot attributes.
- `tlg_plot` module responsible for displaying plots and creating widgets for plot option customizations, which are then dynamically passed to rendering functions.
- `tlg.yaml` file which allows specification of TLG metadata (descriptions, links etc) for compact table display, as well as rendering function and TLG options which are then displayed as shiny widgets and editable.
- Introduces `pkcg01` plot as an example of system implementation.
- Documentation of the system is provided in .github/contributing` folder. 

## Definition of Done

- System for defining TLGs (plots):
  - Allowing the user for quick definition of new plots
  - Easy implementation of variables editable by the user
  - In editable widgets, allow for referencing columns and labels using special key characters _($COLNAME, !LABEL)_
  - Plots rendered using `plotly` for interactivity
  - Documentation available
- First template plot:
  - pkcg01 function implemented and adjusted to fit definition system
  - plots with linear and log scales implemented

## How to test

For basic testing, run the app and **Submit mapping** in the _Mapping and filters_ tab. Then you can go to _TLG_ tab and **Submit order** two default plots. When the plots are rendered, you can test changing the values in the widgets.\

For more advanced testing, please do try to add a new plot. The details are described in `.github/contributing/add-tlg.md`. Please check if the documentation is clear and complete. Adding a new plot will require writing a definition in the `tlg.yaml` file and providing a generating function, which could render a very basic plot.

## Contributor checklist

## Notes to reviewer

Please do review not only the code and functionalities, but also the documentation closely. If anything is confusing, missing or could be explained better - please do let me know. We want the system to be usable by contributors without the need of an introduction on a call.
